projects/rero/ng-core/src/lib/record/record-ui.service.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Invenio angular core
+ * Copyright (C) 2019 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+import { map, mergeMap, delay } from 'rxjs/operators';
+import { of } from 'rxjs';
+
+import { ToastrService } from 'ngx-toastr';
+import { TranslateService } from '@ngx-translate/core';
+
+import { DialogService } from '../dialog/dialog.service';
+import { RecordService } from './record.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RecordUiService {
+  constructor(
+    private dialogService: DialogService,
+    private toastService: ToastrService,
+    private translate: TranslateService,
+    private recordService: RecordService
+  ) { }
+
+  /**
+   * Delete a record by its PID.
+   * @param pid - string, PID to delete
```

Comment:
`@param type` is missing, you can also add `@returns` see: https://jsdoc.app/tags-returns.html

Reply:
done, but the @returns must be set to all functions/methods in project